packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -274,6 +225,26 @@ const BlockInspectorSingleBlock = ( { clientId, blockName } ) => {
 	);
 	const blockInformation = useBlockDisplayInformation( clientId );
 	const borderPanelLabel = useBorderPanelLabel( { blockName } );
+	const contentClientIds = useSelect(
+		( select ) => {
+			// Avoid unnecessary subscription.
+			if ( ! isSectionBlock ) {
+				return;
+			}
+
+			const {
+				getClientIdsOfDescendants,
+				getBlockName,
+				getBlockEditingMode,
+			} = select( blockEditorStore );
+			return getClientIdsOfDescendants( clientId ).filter(
+				( current ) =>
+					getBlockName( current ) !== 'core/list-item' &&
```

Comment:
@youknowriad Having this here hardcoded to only support the list block seems very inflexible 🤔

Is there not more declarative way for us to control this? 

Taking one step back I would love to ensure that the writing mode stuff can properly get interfaced with from custom blocks. 

For example a custom tabs block should still allow adding new tab items inside the tabs... Same with a carousel etc.

Reply:
In other words, this feels like it is patching a hole in our API design of the `contentOnly` mode (#52018) for one special case in core when in reality it is a much larger hole that deserves a more generalized solution 🤔